src/main/java/com/bravo/user/service/UserService.java

```diff
@@ -40,6 +41,24 @@ public UserReadDto create(final UserSaveDto request){
     return resourceMapper.convertUser(user);
   }
 
+  public List<UserReadDto> retrieveByName(
+          final String name,
+          final PageRequest pageRequest,
+          final HttpServletResponse httpResponse){
+    UserFilter filter = new UserFilter();
+    String[] names = name.split(" ");
+
+    filter.setFirstNames(Set.of(names[0]));
+    if(names.length == 2) {
+      filter.setLastNames(Set.of(names[1]));
+    } else if(names.length == 3) {
+      filter.setMiddleNames(Set.of(names[1]));
+      filter.setLastNames(Set.of(names[2]));
+    }
```

Comment:
This is interesting logic. Although, I don't see a scenario where they pass more than 3 names here. Tracing this code it seems like any name after the first 3 would just be ignored.

Reply:
For this particular logic, I went with this approach with an understanding that in a normal scenario, I would have asked for a clarification from the stakeholder for the requirement. From my understanding the API would only take a single value for a name and that name could be any portion of the users full name. If you wanted to be able to provide a more specific search where first, middle, and last name could be provided individually, I would have suggested query parameters or perform a search similar to the POST method for "/retrieve".